### PR TITLE
wallet: watch-only account import improvements

### PR DESCRIPTION
**Pull request by guggero**
This PR fixes two issues that arose when trying to run a purely watch-only wallet and import accounts from xpubs only:

1. Coin selection requires the wallet to be unlocked. Since there is no unlock in watch-only mode, we need to circumvent that check.
2. When importing default accounts, the scope is well-known in advance and we don't need to take the detour via the address type. So we add a new `ImportAccountWithScope` method that both defines the key scope and address schema to use instead of guessing that information from the address type.

cc @Roasbeef

### CODE REVIEW

File: wallet/wallet.go [hunk 1]
```diff
@@ -1169,6 +1169,20 @@ out:
...
+			// selection in that mode.
+			if w.Manager.WatchOnly() {
+				tx, err := w.txToOutputs(
+					txr.outputs, txr.keyScope, txr.account,
+					txr.minconf, txr.feeSatPerKB,
+					txr.coinSelectionStrategy, txr.dryRun,
+				)
+				txr.resp <- createTxResponse{tx, err}
+				continue
+			}
```
Comment (Roasbeef) [inline wallet/wallet.go:1184]: Seems can consolidate the code a bit here w/ something like: 
```go
var release func()
if !w.Manager.WatchOnly() {
    heldUnlock, err := w.holdUnlock()
    .... 
    release := heldUnlock.release()
}

tx, err := w.txToOutputs(
	txr.outputs, txr.keyScope, txr.account,
	txr.minconf, txr.feeSatPerKB,
	txr.coinSelectionStrategy, txr.dryRun,
)

if release != nil {
    release()
}

txr.resp <- createTxResponse{tx, err}
```

Thoughts?

**Review by Roasbeef — CHANGES_REQUESTED** [review]
Nice fix and re-work, one small suggestion here to consolidate the code a bit more in the coin selection loop.

File: wallet/wallet.go (hunk 1 above)
Comment (guggero) [reply wallet/wallet.go]: Great idea, done!

**Comment by guggero** [comment]
Reworked the first commit as suggested.

**Review by Roasbeef — APPROVED** [review]
LGTM 📡

**Pull request merged**